source/Audio.cpp

```diff
@@ -260,6 +266,58 @@ void Audio::Update(const Point &listenerPosition)
 
 
 
+void Audio::UpdateMusic(PlayerInfo &player, Track::GameState state)
+{
+	if(!isInitialized)
+		return;
+
+	if(Preferences::Has("ingame music"))
+	{
+		// If the current playlists conditions are still matching. If not search a new one.
+		bool currentPlaylistValid = currentPlaylist ?
+			currentPlaylist->MatchingConditions(player) : false;
+		if(currentTrack->IsFinished() || !currentPlaylistValid)
+		{
+			if(!currentPlaylistValid)
+			{
+				// If the current playlist is not valid, find a new one based on priority and weight.
+				WeightedList<Playlist *> validPlaylists;
+				int priority = 0;
+				for(auto &playlist : GameData::Playlists())
```

Comment:
I think this loop will result in the first music of each category being selected first
We should probably shuffle them a bit

Reply:
I dont think so, there is a weighted list they are chosen from

Reply:
Oh right its done automatically by the weighted list good then
Its fine then mb just add a comment when yiu do the .get but you dont have to

Reply:
done